File: frontend/portfolioERP-web/src/app/features/orders/models/order-response.ts
import { OrderLineResponse } from './order-line-response';
import { OrderStatus } from './order-status';

export interface OrderResponse {
  id: number;
  orderNumber: string;
  orderDate: string;
  status: OrderStatus;
  customerId: number;
  customerName: string;
  notes: string | null;
  subtotal: number;
  discountAmount: number;
  taxAmount: number;
  totalAmount: number;
  createdAtUtc: string;
  updatedAtUtc: string | null;
  lines: OrderLineResponse[];
}

File: backend/PortfolioERP.Application/Features/Orders/OrderResponse.cs
using PortfolioERP.Domain.Enums;

namespace PortfolioERP.Application.Features.Orders;

public sealed record OrderResponse(
    int Id,
    string OrderNumber,
    DateTime OrderDate,
    OrderStatus Status,
    int CustomerId,
    string CustomerName,
    string? Notes,
    decimal Subtotal,
    decimal DiscountAmount,
    decimal TaxAmount,
    decimal TotalAmount,
    DateTime CreatedAtUtc,
    DateTime? UpdatedAtUtc,
    IReadOnlyList<OrderLineResponse> Lines);